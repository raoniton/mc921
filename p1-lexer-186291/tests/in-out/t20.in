// A simple class
class Calculator {
    int result;

    // Method to add two numbers
    public void add(int a, int b) {
        this.result = a + b;
    }

    // Method to subtract two numbers
    public void subtract(int a, int b) {
        this.result = a - b;
    }

    // Method to display the result
    public void displayResult() {
        print("Result: ", this.result);
    }
}

// Main class
class Main {
    public static void main(String[] args) {
        // Create an instance of Calculator
        Calculator calc = new Calculator();

        // Perform addition
        calc.add(10, 5);
        calc.displayResult(); // Output: Result: 15

        // Perform subtraction
        calc.subtract(10, 5);
        calc.displayResult(); // Output: Result: 5
    }
}