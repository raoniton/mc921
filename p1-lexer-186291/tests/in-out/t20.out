LexToken(CLASS,'class',2,18)
LexToken(ID,'Calculator',2,24)
LexToken(LBRACE,'{',2,35)
LexToken(INT,'int',3,41)
LexToken(ID,'result',3,45)
LexToken(SEMI,';',3,51)
LexToken(PUBLIC,'public',6,91)
LexToken(VOID,'void',6,98)
LexToken(ID,'add',6,103)
LexToken(LPAREN,'(',6,106)
LexToken(INT,'int',6,107)
LexToken(ID,'a',6,111)
LexToken(COMMA,',',6,112)
LexToken(INT,'int',6,114)
LexToken(ID,'b',6,118)
LexToken(RPAREN,')',6,119)
LexToken(LBRACE,'{',6,121)
LexToken(THIS,'this',7,131)
LexToken(DOT,'.',7,135)
LexToken(ID,'result',7,136)
LexToken(ASSIGN,'=',7,143)
LexToken(ID,'a',7,145)
LexToken(PLUS,'+',7,147)
LexToken(ID,'b',7,149)
LexToken(SEMI,';',7,150)
LexToken(RBRACE,'}',8,156)
LexToken(PUBLIC,'public',11,201)
LexToken(VOID,'void',11,208)
LexToken(ID,'subtract',11,213)
LexToken(LPAREN,'(',11,221)
LexToken(INT,'int',11,222)
LexToken(ID,'a',11,226)
LexToken(COMMA,',',11,227)
LexToken(INT,'int',11,229)
LexToken(ID,'b',11,233)
LexToken(RPAREN,')',11,234)
LexToken(LBRACE,'{',11,236)
LexToken(THIS,'this',12,246)
LexToken(DOT,'.',12,250)
LexToken(ID,'result',12,251)
LexToken(ASSIGN,'=',12,258)
LexToken(ID,'a',12,260)
LexToken(MINUS,'-',12,262)
LexToken(ID,'b',12,264)
LexToken(SEMI,';',12,265)
LexToken(RBRACE,'}',13,271)
LexToken(PUBLIC,'public',16,314)
LexToken(VOID,'void',16,321)
LexToken(ID,'displayResult',16,326)
LexToken(LPAREN,'(',16,339)
LexToken(RPAREN,')',16,340)
LexToken(LBRACE,'{',16,342)
LexToken(PRINT,'print',17,352)
LexToken(LPAREN,'(',17,357)
LexToken(STRING_LITERAL,'"Result: "',17,358)
LexToken(COMMA,',',17,368)
LexToken(THIS,'this',17,370)
LexToken(DOT,'.',17,374)
LexToken(ID,'result',17,375)
LexToken(RPAREN,')',17,381)
LexToken(SEMI,';',17,382)
LexToken(RBRACE,'}',18,388)
LexToken(RBRACE,'}',19,390)
LexToken(CLASS,'class',22,407)
LexToken(ID,'Main',22,413)
LexToken(LBRACE,'{',22,418)
LexToken(PUBLIC,'public',23,424)
LexToken(STATIC,'static',23,431)
LexToken(VOID,'void',23,438)
LexToken(MAIN,'main',23,443)
LexToken(LPAREN,'(',23,447)
LexToken(STRING,'String',23,448)
LexToken(LBRACKET,'[',23,454)
LexToken(RBRACKET,']',23,455)
LexToken(ID,'args',23,457)
LexToken(RPAREN,')',23,461)
LexToken(LBRACE,'{',23,463)
LexToken(ID,'Calculator',25,517)
LexToken(ID,'calc',25,528)
LexToken(ASSIGN,'=',25,533)
LexToken(NEW,'new',25,535)
LexToken(ID,'Calculator',25,539)
LexToken(LPAREN,'(',25,549)
LexToken(RPAREN,')',25,550)
LexToken(SEMI,';',25,551)
LexToken(ID,'calc',28,590)
LexToken(DOT,'.',28,594)
LexToken(ID,'add',28,595)
LexToken(LPAREN,'(',28,598)
LexToken(INT_LITERAL,'10',28,599)
LexToken(COMMA,',',28,601)
LexToken(INT_LITERAL,'5',28,603)
LexToken(RPAREN,')',28,604)
LexToken(SEMI,';',28,605)
LexToken(ID,'calc',29,615)
LexToken(DOT,'.',29,619)
LexToken(ID,'displayResult',29,620)
LexToken(LPAREN,'(',29,633)
LexToken(RPAREN,')',29,634)
LexToken(SEMI,';',29,635)
LexToken(ID,'calc',32,699)
LexToken(DOT,'.',32,703)
LexToken(ID,'subtract',32,704)
LexToken(LPAREN,'(',32,712)
LexToken(INT_LITERAL,'10',32,713)
LexToken(COMMA,',',32,715)
LexToken(INT_LITERAL,'5',32,717)
LexToken(RPAREN,')',32,718)
LexToken(SEMI,';',32,719)
LexToken(ID,'calc',33,729)
LexToken(DOT,'.',33,733)
LexToken(ID,'displayResult',33,734)
LexToken(LPAREN,'(',33,747)
LexToken(RPAREN,')',33,748)
LexToken(SEMI,';',33,749)
LexToken(RBRACE,'}',34,776)
LexToken(RBRACE,'}',35,778)
